snuba_sdk/schema.py

```diff
@@ -15,6 +16,9 @@ class ColumnSet:
     def __init__(self, columns: Sequence[Column]) -> None:
         super().__setattr__("column_names", set(c.name for c in columns))
         super().__setattr__("columns", set(columns))
+        super().__setattr__(
+            "required_columns", set([r for r in self.columns if r.required])
+        )
```

Comment:
Unrelated to this PR but I think it's worth raising. 

We have all this fancy code to set stuff on the class but we wouldn't need it if the dataclass was not frozen. Why do we mark it as frozen and then mutate it?